src/catch.rs

```diff
@@ -0,0 +1,92 @@
+use std::marker::PhantomData;
+
+use jni::{
+    objects::{AutoLocal, JObject},
+    JNIEnv,
+};
+
+use crate::{cast::Upcast, error::Error, java::lang::Throwable, JvmOp, Local};
+
+/// Plumbing utility to check the exception state of the JVM thread convert it into a [`crate::Result`].
+// XX: many of the jni crate checked methods do this automatically. We only need this if/when we invoke
```

Comment:
I'm confused -- those methods don't, presumably, convert to `Error::Thrown`, right, but rather to some JNI-internal result variant?

Reply:
I think we can delete this fn, but for posterity, the `jni` crate methods automatically check for an exception flag and then return a bare `jni::errors::Error::JavaException` without storing a reference to the exception object.

The bigger issue I was worried about was proceeding to make further Java calls without first checking and clearing that flag. It would cause the exception to be rethrown in an odd place. But, it looks like the `jni` crate has put this check everywhere.